### DISCUSSION TITLE
Recommended Tabular Renderer?

### DISCUSSION THREAD

**Discussion by iwelch** [mojolicious/mojo#2022] [Q&A] [answered]
After many years, I need to write another cgi script again.  I am working with tabular data (a rectangular array) that I want to display in my html template.  Is there a recommended solution for this that also adds niceties (e.g., sorting) or should I roll what I need myself?

**Comment by MartinMcGrath** [comment, marked as answer]
I use [DataTables](https://datatables.net/), feeding it JSON data from my Mojo app.

**Reply by iwelch** [reply]
easy to drop a simple example?  (if not, don't.)

**Reply by MartinMcGrath** [reply]
> easy to drop a simple example? (if not, don't.)

As outlined on the homepage, the quickest way to test would be to have your template include the CSS/JS files and render an HTML table (from DB query or whatever) named 'myTable', and add the JS code:
```
$(document).ready( function () {
    $('#myTable').DataTable();
} );
```

If it fits your requirements you could investigate the various options provided. The thorough docs provide loads of [examples](https://datatables.net/examples/index).
